app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -16,20 +31,83 @@ def save_ems_container_inventory(ems, hashes, target = nil)
     ems.save!
   end
 
-  def save_container_projects_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_projects.reset
-    deletes = if target.kind_of?(ExtManagementSystem)
-                :use_association
-              else
-                []
-              end
+  def initialize_inventory_collections(ems)
+    @inv_collections = {}
+    @inv_collections[:container_projects] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerProject,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_projects
+    )
+    @inv_collections[:container_quotas] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuota,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_quotas,
+      #:arel => ContainerQuota.joins(:container_project).where(:container_projects => {:ems_id => ems.id}),
+    )
+    @inv_collections[:container_quota_items] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuotaItem,
+      :parent => ems,
+      :association => :container_quota_items,
+      #:arel => ContainerQuotaItem.joins(:container_quota => :container_project).where(:container_projects => {:ems_id => ems.id}),
+      :manager_ref => [:container_quota, :resource],
+    )
+    @inv_collections[:container_nodes] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerNode,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_nodes,
+    )
+    # ContainerCondition is polymorphic child of ContainerNode & ContainerGroup
+    @inv_collections[:container_node_conditions] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerCondition,
+      :parent => ems,
+      :association => :container_node_conditions,
+      #:arel => ContainerCondition.joins(
+      #  'INNER JOIN "container_nodes" ON "container_conditions"."container_entity_id" = "container_nodes"."id"'
+      #).where(:container_entity_type => 'ContainerNode',
+      #        :container_nodes => {:ems_id => ems.id}),
+      :manager_ref => [:container_entity, :name],
+    )
+
+    # polymorphic child of ContainerNode & ContainerImage
+    # TODO these are dumb, find way to share same InventoryCollections via both.
```

Comment:
ems_id everywhere would be nice.  (BTW, one reason tags special saving code got messy is because it can't be partitioned by ems_id.  all share same tag rows.)

I don't have a native ems_ref for ComputerSystem.  It's not an entity in kubernetes, we just parse some of the k8s data into a separate table.  well since it's 1:1, I could maybe just use parent node's / image's GUID as ComputerSystem's ems_ref.  not sure I like synthetic ems_refs...

now let's say we added to ComputerSystem just ems_id without ems_ref.  tempting because avoids 
ems_ref "abuse".
then imagine a targeted refresh on a single node.  it should replace the node's ComputerSystem(s)¹, but only that node's.  So if we simplify the arel to be "all ComputerSystem(s) with this ems_id", we'd give up deletions in some cases :-(

¹ bad example as it just has_one, but say ContainerQuotaItem(s) don't have ems_ref and are many.